### PR TITLE
Inline the track-wide exercise README insert into the exercise template

### PR DESCRIPTION
**Pull request by kytrinyx**
Since we now have a track-wide exercise template, I've inlined the markdown README insert in the docs directory into the template.

The first three commits do general cleanup so that the diff when inlining is minimal.

After inlining, I've got a commit that tweaks the inlined text to make it refer to the actual exercise directory name, since the template has access to that value at generation time.

The final commit regenerates all the READMEs using the updated template.

See https://github.com/exercism/meta/issues/94

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
Looking good! We as the C# track's maintainers could have a discussion to determine if the track-specific bit about the submit directory adds any real value. @robkeim @jpreese what do you think?

**Comment by robkeim** [comment]
Thanks @kytrinyx! @ErikSchierboom I'm not sure how much additional value that submit directory is adding and if we felt there was a lot of value add I'd argue we should try to get it integrated for all of the tracks. What are your thoughts?

**Comment by ErikSchierboom** [comment]
I don't think it adds much value. Personally, I'm in favor of just removing it. The Exercism CLI should actually help the user with these kinds of errors I feel.

**Comment by kytrinyx** [comment]
> The Exercism CLI should actually help the user with these kinds of errors I feel.

Agreed! I'll remove it and push this back up.

**Comment by kytrinyx** [comment]
Oh. Didn't realize this was merged. I'll submit a new PR :)

**Comment by kytrinyx** [comment]
Actually, I'll let you all sort it out now that I think of it. Sorry for the noise 🌷

**Comment by ErikSchierboom** [comment]
@kytrinyx One question: is there a tool with which we can regenerate the docs?

**Comment by kytrinyx** [comment]
> is there a tool with which we can regenerate the docs?

Shoot, yes! 100% absolutely yes :)

TL;DR:

```
configlet generate . --spec-path=../path/to/problem-specifications
```

Long version:
https://github.com/exercism/docs/blob/master/maintaining-a-track/regenerating-exercise-readmes.md

**Comment by ErikSchierboom** [comment]
Wonderful, thanks!
